File: 23-09-26/3.c

#include <stdio.h>

//the length of consecutive 1 counter
int runLength(int a[], int n, int i)
{
// returning length 0 as there is no 1
if (a[i] == 0)
return 0;

int count = 0;  

//this will check the one repeating for the remaining part of array frmo i and then returns l
//
while (i < n && a[i] == 1)  
{  
    count++;  
    i++;  
}  

return count;

}

int main()
{
// Example data
int n;
int k;

printf("Enter The value of n: ");
scanf(" %d",&n);
printf("Enter the value of k: ");
scanf(" %d",&k);

int a[n];

printf("Enter The vector: \n");
for(int i =0;i<n;i++){
	scanf(" %d",&a[i]);
}

int answer = 0;  

for (int i = 0; i < n; i++)  
{  
    int length = runLength(a, n, i);  
//then going to i+1 if the folloeing below beak condition doesnt satisfy.
//
//and then for each and every i , we check the length 
    if (length > k)  
    {  
        // The violation occurs at the kth+1  
        // studying session of this run.  
        answer = i + k + 1;  
        break;  
    }  
}  

printf("n = %d\n", n);  
printf("k = %d\n", k);  

printf("Entries: ");  
for (int i = 0; i < n; i++)  
{  
    printf("%d ", a[i]);  
}  

printf("\nOutput = %d\n", answer);  

return 0;

}

File: 23-09-26/3.py
import numpy as np

def runLength(a, i):
    if a[i] == 0:
        return 0

    count = 0

    while i < len(a) and a[i] == 1:
        count += 1
        i += 1

    return count


a = np.array(list(map(int, input("Enter the array: ").split())))

k = int(input("Enter k: "))

n = len(a)
answer = 0

for i in range(n):
    length = runLength(a, i)

    if length > k:
        answer = i + k + 1
        break

print(f"n = {n}")
print(f"k = {k}")

print("Entries:", *a)

print(f"Output = {answer}")
